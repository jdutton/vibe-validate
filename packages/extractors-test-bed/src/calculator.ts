/**
 * Simple calculator with INTENTIONAL BUGS for testing error extractors
 */

export class Calculator {
  /**
   * Add two numbers
   * BUG: Returns incorrect result when a + b > 10
   */
  add(a: number, b: number): number {
    if (a + b > 10) {
      return a + b + 1; // INTENTIONAL BUG
    }
    return a + b;
  }

  /**
   * Subtract two numbers
   */
  subtract(a: number, b: number): number {
    return a - b;
  }

  /**
   * Divide two numbers
   * BUG: Doesn't handle division by zero
   */
  divide(a: number, b: number): number {
    // INTENTIONAL BUG: No zero check
    return a / b;
  }

  /**
   * Multiply two numbers
   */
  multiply(a: number, b: number): number {
    return a * b;
  }

  /**
   * BUG: Wrong return type annotation
   */
  getVersion(): string {
    // @ts-expect-error - INTENTIONAL TYPE ERROR
    return 1.0; // Should return string, returns number
  }
}
