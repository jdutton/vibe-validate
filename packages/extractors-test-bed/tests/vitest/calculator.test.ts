/**
 * Vitest tests with INTENTIONAL FAILURES for testing error extractors
 *
 * This test suite mirrors the Jest tests to demonstrate cross-framework
 * error extraction. Same failures, different output format.
 */

import { describe, it, expect, beforeEach } from 'vitest';
import { Calculator } from '../../src/calculator.js';

describe('Calculator (Vitest)', () => {
  let calc: Calculator;

  beforeEach(() => {
    calc = new Calculator();
  });

  describe('Addition', () => {
    it('should add two small numbers correctly', () => {
      // INTENTIONAL FAILURE: Calculator.add() has a bug when sum > 10
      const result = calc.add(6, 7);
      expect(result).toBe(13); // Expected: 13, Actual: 14 (bug)
    });

    it('should handle negative numbers', () => {
      const result = calc.add(-5, 3);
      expect(result).toBe(-2);
    });
  });

  describe('Division', () => {
    it('should divide numbers correctly', () => {
      const result = calc.divide(10, 2);
      expect(result).toBe(5);
    });

    it('should throw error when dividing by zero', () => {
      // INTENTIONAL FAILURE: Calculator.divide() doesn't handle division by zero
      expect(() => calc.divide(10, 0)).toThrow('Division by zero');
      // This will fail because divide() doesn't throw an error
    });
  });

  describe('Multiplication', () => {
    it('should multiply numbers', () => {
      // INTENTIONAL FAILURE: Wrong expected value
      const result = calc.multiply(4, 5);
      expect(result).toBe(25); // Expected: 25, Actual: 20
    });
  });

  describe('Type Errors', () => {
    it('should return version as string', () => {
      const version = calc.getVersion();
      // INTENTIONAL FAILURE: getVersion() returns number, not string
      expect(typeof version).toBe('string');
    });
  });

  describe('Runtime Errors', () => {
    it('should read config file', async () => {
      // INTENTIONAL FAILURE: File does not exist (ENOENT)
      const fs = await import('node:fs/promises');
      const config = await fs.readFile('/this/file/does/not/exist.json', 'utf8');
      expect(config).toBeDefined();
    });
  });

  describe('Timeout', () => {
    it('should complete within timeout', async () => {
      // INTENTIONAL FAILURE: Timeout
      await new Promise(resolve => setTimeout(resolve, 100000)); // Takes 100 seconds
      expect(true).toBe(true); // Unreachable - test will timeout before this
    }, 10); // 10ms timeout
  });

  describe('Snapshot Mismatch', () => {
    it('should match snapshot', () => {
      // INTENTIONAL FAILURE: Snapshot will not match
      const result = {
        status: 'failed',
        timestamp: Date.now(),
        errors: ['Something went wrong'],
      };
      expect(result).toMatchSnapshot();
    });
  });
});
