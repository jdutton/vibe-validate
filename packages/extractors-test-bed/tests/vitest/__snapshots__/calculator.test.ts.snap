// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Calculator (Vitest) > Snapshot Mismatch > should match snapshot 1`] = `
{
  "errors": [
    "Something went wrong",
  ],
  "status": "failed",
  "timestamp": 1761317867279,
}
`;
